wire-ios-data-model/Source/MLS/MLSActionExecutor.swift

```diff
@@ -248,30 +252,31 @@ actor MLSActionExecutor: MLSActionExecutorProtocol {
                 )
 
             case .removeClients(let clients):
-                return try coreCrypto.perform {
-                    try $0.removeClientsFromConversation(
-                        conversationId: groupID.bytes,
+                return try await coreCrypto.performAsync {
+                    try await $0.removeClientsFromConversation(
+                        conversationId: groupID.data,
                         clients: clients
                     )
                 }
 
             case .updateKeyMaterial:
-                return try coreCrypto.perform {
-                    try $0.updateKeyingMaterial(conversationId: groupID.bytes)
+                return try await coreCrypto.performAsync {
+                    try await $0.updateKeyingMaterial(conversationId: groupID.data)
                 }
 
             case .proposal:
-                guard let bundle = try coreCrypto.perform({ try $0.commitPendingProposals(
-                    conversationId: groupID.bytes
+                guard let bundle = try await coreCrypto.performAsync({ try await $0.commitPendingProposals(
+                    conversationId: groupID.data
+
                 ) }) else {
                     throw Error.noPendingProposals
                 }
 
                 return bundle
 
-            case .joinGroup(let groupInfo):
-                let conversationInitBundle = try coreCrypto.perform { try $0.joinByExternalCommit(groupInfo: groupInfo.bytes,
-                                                                                                  customConfiguration: .init(keyRotationSpan: nil, wirePolicy: nil), credentialType: .basic) }
+            case .joinGroup(let publicGroupState):
+                let conversationInitBundle = try await coreCrypto.performAsync { try await $0.joinByExternalCommit(groupInfo: publicGroupState,
```

Comment:
`Nitpick`: formatting could be better

Reply:
This code has since been refactored in another PR.